lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1299,26 +1299,61 @@ getNetworkInformation (_block0, bp, st) nl = do
                 }
         }
   where
-    sp :: W.SlotParameters
-    sp = W.SlotParameters
-        (bp ^. #getEpochLength)
-        (bp ^. #getSlotLength)
-        (bp ^. #getGenesisBlockDate)
-        (bp ^. #getActiveSlotCoefficient)
+    sp = toSlotParameters bp
 
     -- Unsafe constructor for the next epoch. Chances to reach the last epoch
     -- are quite unlikely in this context :)
     unsafeEpochSucc :: HasCallStack => W.EpochNo -> W.EpochNo
     unsafeEpochSucc = fromMaybe bomb . W.epochSucc
       where bomb = error "reached final epoch of the Blockchain!?"
 
+toSlotParameters :: BlockchainParameters -> W.SlotParameters
+toSlotParameters bp = W.SlotParameters
+    (bp ^. #getEpochLength)
+    (bp ^. #getSlotLength)
+    (bp ^. #getGenesisBlockDate)
+    (bp ^. #getActiveSlotCoefficient)
+
 getNetworkParameters
-    :: forall t. ()
-    => (Block, BlockchainParameters, SyncTolerance)
-    -> NetworkLayer IO t Block
+    :: (Block, BlockchainParameters, SyncTolerance)
     -> ApiEpochNumber
     -> Handler ApiNetworkParameters
-getNetworkParameters _ _ _ = throwError err501
+getNetworkParameters (_block0, bp, _st) apiEpochNum = do
+    case apiEpochNum of
+        ApiEpochNumberInvalid txt ->
+            liftHandler $ throwE $ ErrGetNetworkParametersInvalidValue txt
```

Comment:
Were there any specific problem that caused you to add `ApiEpochNumberInvalid`? 

I would have thought we could get by without it, and that `ErrGetNetworkParametersInvalidValue` (or at least the HTTP code) could be thrown via the `FromJSON` instance.

Reply:
I have in task specification : 
```
400 # When the epoch parameter is malformed
404 # When the epoch is valid doesn't exist
```
which is reasonable. We want either `latest` or epoch number `0` or `225` So if we put here `-1` or `aaaa` then it is invalid value and I want to emit `err400`.

But we can have also legal value but bigger than current epoch number and we want to differentiate this case from the above. And want to have this : 
```
[pawel@arch ~]$ curl http://localhost:8090/v2/network/parameters/2000000 --http1.1
{"code":"not_such_epoch_no","message":"I couldn't show blockchain parameters for epoch number later than current one. You requested 2000000 epoch. Current one is 1196747. Use smaller epoch than current or 'latest'."}
```
So, what I did I moved responsibility from aeson parser to Server and here I am validating this and properly (and in controlled way) handle it

Reply:
> what I did I moved responsibility from aeson parser to Server and here I am validating this and properly (and in controlled way) handle it

Ok thanks, want to have a look and understand this a bit more tomorrow